tests/python/pants_test/option/test_options.py

```diff
@@ -969,11 +969,7 @@ def test_middle_scoped_options(self):
                           })
     self.assertEquals(100, options.for_global_scope().a)
     self.assertEquals(99, options.for_scope('compile').a)
-
-    # TODO(John Sirois): This should pick up 99 from the the recursive global '--a' flag defined in
-    # middle scope 'compile', but instead it picks up `a`'s value from the config DEFAULT section.
-    # Fix this test as part of https://github.com/pantsbuild/pants/issues/1803.
-    self.assertEquals(100, options.for_scope('compile.java').a)
```

Comment:
Yes; mentioned in description: rather than preferring a default value in a higher priority file, this will now take an explicitly specified value in a lower priority file before using defaults.